internal/pkg/earpointer/state.go

```diff
@@ -308,10 +313,27 @@ func (state *state) ToPartitions() *Partitions {
 // After this call, internal data structures are optimized for lookups only,
 // and no mutations should be made.
 func (p *Partitions) finalize() {
-	// Construct the map from field references to their parents.
-	p.constructFieldParentMap()
-
-	// Construct the members. members maps partitions representatives to the
+	// (1) Construct a new parent map that maps a reference to its representative
+	//     so as to accelerate representative lookup.
+	// For example, supposed the old parent map contains r1->r2, r2->r3,
+	// r3->r4(representative), then the new parent map contains r1->r4.
+	// "r4" can be returned immediately without chasing the union-find relation
```

Comment:
Won't chasing only occur on lookup? And then because of compression it won't happen again?

Reply:
Yes chasing happens when an EAR state is constructed:

func (state *state) representative(ref Reference) Reference {
	failureCallback := func(ref Reference) Reference {
		// Some global variables has no declarations in the SSA;
		// Insert their references dynamically.
		state.Insert(ref)
		return ref
	}
	return state.lookupPartitionRep(ref, failureCallback)
}

After the entire state has been constructed, we can accelerate the lookup and avoid chasing:

func (p *Partitions) Representative(ref Reference) Reference {
	return p.parents[ref]
}

Since "Representative()" is a popular operation, it is worth to optimizing it here.